.github/workflows/codeql.yaml

```diff
@@ -3,8 +3,12 @@
 name: "Run CodeQL security analysis"
 
 on:
-  workflow_dispatch:
-  workflow_call:
+  push:
+    branches: ["main"]
+  pull_request:
+    branches: ["main"]
+  schedule:
+    - cron: "34 19 * * 0"
```

Comment:
Random? 
![image](https://github.com/user-attachments/assets/3095f11d-246e-4c21-bbb4-f2057eb65056)

Reply:
Ah, you just copied it from the old one. Alright.

Reply:
But if it _needs_ to happen on push, maybe we shouldn't make it run periodically? Won't it just fail?

Reply:
I guess it can be commented out and activated as per need in a project.

Reply:
It just needs an on push hook to run properly. Dont think it needs to actually run on push ...

Reply:
That's super-weird, if true.